### PR TITLE
WIP keep \t escaped inside strings

### LINKED ISSUE #1695: \t in string replaced by ASCII 9
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAHATgSwHYBcAzXMAHUgAlEAba2U0-AdVk2vnMgBpwoBnRPnx4A5n1RgA2rwigI8qABJ8AT3TI0kPPm4z5kRQDcAhtQCuGqXoUAGHgoWQAkriQEAytgBeyew4gALNZgALp6AL5%2BssEGquoSWgS6-gYm5pbS-hAAjFEOkACyxgAeADJ4iKWIuCL4ABbJWdkATDbBYQqRenL5ymqWkABGsLDUjY5GphYSmVnNeY7uiAC22HB0uACC%2BACirgDyhOW4vsEQhKYC7RF5PRNxA8Oj4-qT6TNnYADMC-ru6MZoIgAEKIQisRAABWMmGMy0EiEwLwU%2BEwFmunVuMT68U0TzGv1SUwynwCvwgkH%2BgJBYIhpVgAHdEdBjAIXIZYCzhLBcMj5Kj0f4OvIugo7q8Hgl8Xyie80LN-ABWclQKlA0HgzCIACq6HUmBZbNwHK52B5fPOl0QGJFWJSOMeIwJ2LS03lnwAbCrKQD1bStQBhaisvgBnl8AXQfCsC1gC7UK5Cm7dbGSvFOmVvN1WLJgADs3rVNM1iAKK0GiNj8cTDmFEFF8nFFIdUozhKzJNzAA5C77ixCwxsq1abfW7b001Bpe3XZ2sgBOXvUzaEfCIsPLZbGWMC61JzEp%2B2ToZtl3Ej657J2T4%2B6kaiFLVbrc0q6t72vJsWp-qt54z8-upeuQ3kWK5rpgj5rKML6fLuo5gA20RHj%2B6Z-meco5k08w3i4bj4PsuAACqYCoTDYPUw4Ju%2BCh1gh473ChU6nvas4Xk0PwgX2myYLAZiuAAIjQ2CrOBO5otR8i0YhYBNkox7aJmrGAU0ZI3kUxROIQhF1NUOxUe4dSsMw2DwBRKoBG0%2B62oeE6MYkOj-hhCoONkypqSULiENgxT7PqxjRpgOzFFgiB8HwZq8iqiqWR%2BB5fshuJQApjnZs5CjZF67nFAASogcCYPAZGmQ05kxTRn6Nt%2BiX2YpAGYf42QFlluX5fAAByZjLBWmCHE4a7LOIKrZPB0mybEdnLGY1DCNQFQAGKsFuQiIoRP4pXODU9jeLWsPABRTTN82Lf5onetAdQwoC4EAPoMiZFEjfREp2cl6GpZ82SLll3Gwio%2ByYOUEZFWZpJlZJFVIbZ1WvSxdVpfIrTeupP3GH9APYBGHVdYivX9YNH2PTZDHVZN03YLNJwLZgS3gat8TrWx-jNMBuaQCjaOA-g%2B1kxTiBUzTlZnRdsJRoit33Q0hPxVDAww70Sn1Q4zTYaz6kAGrEsCeDwKIwMlaDUuVQlstJAzylMxxqslHNfFRhFWuuLrEuxhZhuQ8TJsOW9G1K6pVvFPxsD4AA4oIQUhWFEV67G0VuzJVWOmhsNOZ8zRuarB3kxUwJ0NAADWwIi3ngh8AR7hwogg6dZFnxvnHY0tqhzrJ%2B9ubNJlrNtYgDK86C%2BBMtUdOIIJXm4OREWbK4ZbY5g%2BO5nXVljkTz3VdO3uM0rTWswA0ogiDoJp2nVC45fwscpzzyOi90cvzbyab6-m0rW3%2B3pqx8HU0elfXCe-s38tw1Tl9VmngajUEQNxEQnVqjMEQIMAoCAL5ZAXrFay0sPZ-1qinXMXxrys02LNEQuAba4DtjyTwRD-JmC1IRWAuEYH%2BQipRGs5U4pGxlpgs2isFBfBZlkNm01ES4EYYYSoPIRDT26iPPA49wzMIkhAKST0752TXi3H2PCVb8O5sIc%2BpQ4SDHgMYIMsBwo1C7j3Co8if7G04Y-bh8gviW34fxDGxhBjgLfhjOo7gVAEBKNY6%2Bo1f6aGqPAQ4593CqHAVg1uWQvh%2B34XscJRwrFnTYIQSANiOFN1iRoxx6d%2BG733vQggLhC7GFIfrS%2BVFskYNyVw%2BGEAvgd34TnSpedz58G4rxVwXcIyID2pnXm4ctSRzkSqOCQTlFyVUcxAB2D4lb34VEnAUYEFIECag%2BsvAQh%2BC0HwOa4UJB13CEAA)

#### Code

```fsharp
printfn "Hello\tWorld"
```
            

#### Result

```fsharp
printfn "Hello	World"

```
            
#### Problem description

The string is not recognized as trivia so the string from the AST is printed back out (thx @nojaf for quick analysis).
Thus, the `\t` in the string is replaced by ASCII 9 (TAB).

via ionide/ionide-vscode-fsharp#1534

#### Extra information

- [ ] The formatted result breaks by code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas Master at 05/01/2021 14:37:19 - d8e76d456b7eda5419e5eb836a793fa5c6ab491a

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>
        
Edit: credits

### PR DESCRIPTION
**Pull request by stanciuadrian**
Closes #1695.
WIP

### CODE REVIEW

File: src/Fantomas.Tests/SynConstTests.fs [hunk 1]
```diff
@@ -301,13 +301,13 @@ let ``hex escape in string literal should be preserved, 1508`` () =
...
         """let hexEscape = "\x00"
-let controlEscapes = "\a \b \f \v"
+let controlEscapes = "\a \b \f \t \v"
 """
         config
     |> should
         equal
         """let hexEscape = "\x00"
-let controlEscapes = "\a \b \f \v"
+let controlEscapes = "\a \b \f \t \v"
```
Comment (nojaf) [inline src/Fantomas.Tests/SynConstTests.fs:310]: Could you please write a new test based on your original report?
The other characters in the string already make the string trivia kick in.
So this test would indeed pass without updating the regex.

**Review by nojaf — CHANGES_REQUESTED** [review]

File: src/Fantomas.Tests/SynConstTests.fs (hunk 1 above)
Comment (stanciuadrian) [reply src/Fantomas.Tests/SynConstTests.fs]: Done

**Review by nojaf — APPROVED** [review]
Thanks!

**Comment by stanciuadrian** [comment]
Thanks for the mentoring.
It's a friendly project for first-timers.

**Comment by nojaf** [comment]
Those are kind words, my gratitude!
If you want to learn more about developing Fantomas, I'd launched [a video series](https://www.youtube.com/playlist?list=PLvw_J2kfZCX3Mf6tEbIPZXbzJOD1VGl4K) a while ago.
The main idea is that it should be possible to solve your bugs giving some guidance. Happy to see this in practise!

**Pull request merged**